scapy/layers/inet.py

```diff
@@ -316,12 +318,30 @@ class IPOption_SDBM(IPOption):
      "Timestamp": 8,
      "AltChkSum": 14,
      "AltChkSumOpt": 15,
+     "MD5": 19,
      "Mood": 25,
      "UTO": 28,
+     "AO": 29,
      "TFO": 34,
      })
 
 
+class TCPAOValue(Packet):
+    """Value of TCP-AO option"""
+    fields_desc = [
+        ByteField("keyid", None),
+        ByteField("rnextkeyid", None),
+        StrLenField("mac", "", length_from=lambda p:len(p.original) - 2),
+    ]
+
+
+def get_tcpao(tcphdr):
```

Comment:
Please add a doctring

Reply:
OK.

Also added a type annotation and fixed some of the type annotations that I added after the docstring (they must be before). They were missed because mypy is not enabled for inet/inet6 so no warnings are printed.